mlir/lib/Dialect/ArmSVE/Transforms/LegalizeVectorStorage.cpp

```diff
@@ -145,8 +154,10 @@ struct LegalizeSVEMaskAllocation : public OpRewritePattern<AllocLikeOp> {
   }
 };
 
-/// Replaces vector.type_casts of unrealized conversions to illegal memref types
-/// with legal type casts, followed by unrealized conversions.
+/// Replaces vector.type_casts of unrealized conversions to SVE predicate memref
+/// types that are _illegal_ to load/store from (!= svbool_t), with type casts
+/// of memref types that are _legal_ to load/store, followed by unrealized
+/// conversions [1][2].
```

Comment:
Sorry for being pedantic, but you'd normally put a reference next to the thing that the reference defines. So this:
```
/// Replaces vector.type_casts of unrealized conversions to SVE predicate memref
/// types that are _illegal_ to load/store from (!= svbool_t [1]), with type casts
/// of memref types that are _legal_ to load/store, followed by unrealized
/// conversions.
```
Rather than:
```
/// Replaces vector.type_casts of unrealized conversions to SVE predicate memref
/// types that are _illegal_ to load/store from (!= svbool_t), with type casts
/// of memref types that are _legal_ to load/store, followed by unrealized
/// conversions [1][2].
```

As "SVE mask" is not referred to in this snippet, I'd just skip that reference.